File: DailyNews/src/main/java/com/jw/dailyNews/adapter/ViewPagerAdapter.java
package com.jw.dailyNews.adapter;

import android.support.v4.view.PagerAdapter;
import android.view.View;
import android.view.ViewGroup;

import java.util.List;

/**
 * 创建时间：
 * 更新时间：2017/11/11 0011 上午 12:28
 * 作者：Mr.jin
 * 描述：图片新闻viewpager适配器
 */
public class ViewPagerAdapter extends PagerAdapter 
{
	private List<View> views;
	public ViewPagerAdapter(List<View> views) {
		this.views = views;
	}

	@Override
	public int getCount() {
		return this.views.size();
	}

	public Object instantiateItem(ViewGroup container, int position) {
		 container.addView(views.get(position));
		return views.get(position);
	}

	@Override
	public boolean isViewFromObject(View view, Object obj) {
		return view==obj;
	}
	public void destroyItem(ViewGroup container, int position, Object object) {
		 container.removeView(views.get(position));
	}
}

File: DailyNews/src/main/java/com/jw/dailyNews/adapter/ViewPagerAdapter.kt
package com.jw.dailyNews.adapter

import android.support.v4.view.PagerAdapter
import android.view.View
import android.view.ViewGroup

/**
 * 创建时间：
 * 更新时间：2017/11/11 0011 上午 12:28
 * 作者：Mr.jin
 * 描述：图片新闻viewpager适配器
 */
class ViewPagerAdapter(private val views: List<View>) : PagerAdapter() {

    override fun getCount()=this.views.size

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        container.addView(views[position])
        return views[position]
    }

    override fun isViewFromObject(view: View, obj: Any)=view === obj

    override fun destroyItem(container: ViewGroup, position: Int, `object`: Any) {
        container.removeView(views[position])
    }
}